explain-partitions.md

```diff
@@ -0,0 +1,132 @@
+ ('2020-06-01', RANDOM_BYTES(1024), RANDOM_BYTES(1024), RANDOM_BYTES(1024)),
+ ('2020-09-01', RANDOM_BYTES(1024), RANDOM_BYTES(1024), RANDOM_BYTES(1024));
+
+INSERT INTO t1 SELECT NULL, a.d, RANDOM_BYTES(1024), RANDOM_BYTES(1024), RANDOM_BYTES(1024) FROM t1 a JOIN t1 b JOIN t1 c LIMIT 10000;
+INSERT INTO t1 SELECT NULL, a.d, RANDOM_BYTES(1024), RANDOM_BYTES(1024), RANDOM_BYTES(1024) FROM t1 a JOIN t1 b JOIN t1 c LIMIT 10000;
+INSERT INTO t1 SELECT NULL, a.d, RANDOM_BYTES(1024), RANDOM_BYTES(1024), RANDOM_BYTES(1024) FROM t1 a JOIN t1 b JOIN t1 c LIMIT 10000;
+INSERT INTO t1 SELECT NULL, a.d, RANDOM_BYTES(1024), RANDOM_BYTES(1024), RANDOM_BYTES(1024) FROM t1 a JOIN t1 b JOIN t1 c LIMIT 10000;
+
+SELECT SLEEP(1);
+ANALYZE TABLE t1;
+```
+
+The following example shows a statement against the newly created partitioned table:
+
+{{< copyable "sql" >}}
+
+```sql
+EXPLAIN SELECT COUNT(*) FROM t1 WHERE d = '2017-06-01';
+```
+
+```sql
++------------------------------+---------+-----------+---------------------------+-------------------------------------------+
+| id                           | estRows | task      | access object             | operator info                             |
++------------------------------+---------+-----------+---------------------------+-------------------------------------------+
+| StreamAgg_21                 | 1.00    | root      |                           | funcs:count(Column#8)->Column#6           |
+| └─TableReader_22             | 1.00    | root      |                           | data:StreamAgg_10                         |
+|   └─StreamAgg_10             | 1.00    | cop[tikv] |                           | funcs:count(1)->Column#8                  |
+|     └─Selection_20           | 8.87    | cop[tikv] |                           | eq(test.t1.d, 2017-06-01 00:00:00.000000) |
+|       └─TableFullScan_19     | 8870.00 | cop[tikv] | table:t1, partition:p2017 | keep order:false                          |
++------------------------------+---------+-----------+---------------------------+-------------------------------------------+
+5 rows in set (0.01 sec)
+```
+
+Starting from the inner-most (`└─TableFullScan_19`) operator and working back towards the root operator (`StreamAgg_21`):
+
+* TiDB successfully identified that only one partition (`p2017`) needed to be accessed. This is noted under `access object`.
+* The partition itself was scanned in the operator `└─TableFullScan_19` and then `└─Selection_20` was applied to filter for rows that have a start date of `2017-06-01 00:00:00.000000`.
+* The rows that match `└─Selection_20` are then stream aggregated in the coprocessor, which natively understands the `count` function.
+* Each coprocessor request then sends back one row to  `└─TableReader_22` inside TiDB, which is then stream aggregated under `StreamAgg_21` and one row is returned to the client.
+
+The following example shows partition pruning has not applied:
```

Comment:
The expression is a little bit confusing to me: "has not been applied" or "has not applied sth"?

Reply:
I think you are right, it should be clearer - since arguably partition pruning has been applied but rather is not effective. How about `In the following example, partition pruning does not eliminate any partitions:`

Reply:
Sounds better to me.